src/main/java/mshantadze/pages/base/BasePage.java

```diff
@@ -4,32 +4,35 @@
 import mshantadze.utils.ui.UIElementWrapper;
 import org.openqa.selenium.WebDriver;
 import org.openqa.selenium.WebElement;
+import org.openqa.selenium.remote.DesiredCapabilities;
+import org.openqa.selenium.remote.RemoteWebDriver;
 import org.openqa.selenium.support.ui.ExpectedConditions;
 import org.openqa.selenium.support.ui.WebDriverWait;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.testng.Assert;
 
+import java.net.MalformedURLException;
+import java.net.URL;
 import java.time.Duration;
 
 public abstract class BasePage extends UIElementWrapper {
     private static final Logger LOGGER = LoggerFactory.getLogger(BasePage.class);
-    private static final String url = ConfigService.init().get("url");
-    protected WebDriver driver;
+    private final ConfigService configService = ConfigService.init();
+    private final String url = configService.get("url");
 
     protected BasePage (WebDriver driver) {
         super(driver);
-        this.driver = driver;
     }
 
-    public void verifyThatPageIsLoaded() {
+    public void verifyThatPageIsLoaded(WebDriver driver) {
```

Comment:
Why do you need Driver as parameter here. Make field WebDriver in BasePage and set driver with constructor